### PR TITLE
Implement XSS safety - External links on Project pages are now rendered in Markdown

### LINKED ISSUE #5138: XSS Vulnerability on Project page - External Project Links
## Security Issue
There is a cross-site scripting vulnerability on the Project page which allows JavaScript code to be injected into the External Links section.

screenshot 2018-12-11 at 11 01 01
_In the project https://www.zooniverse.org/projects/camallen/my-testing-project, the URL to the Blog is `javascript:alert('test test test test')` - clicking on it executes the alert code, which it shouldn't be allowed to. Confirmed on OSX+Chrome._

We are unsure if other `<a>` Anchors in PFE have similar vulnerabilities, so we'll need to look into that too.

### Analysis
This is a basic XSS vulnerability - in this case, the `<a>` tags don't clean the user-supplied content for its `href` attribute.

As suggested by @eatyourgreens , changing these Anchors into Markdown elements may be an easy way to fix the problem, as Markdown perform the necessary sanitisation of the input.

### Status
Security priority, needs fixing.

### PR DESCRIPTION
**Pull request by shaunanoordin**
## PR Overview
This PR fixes an XSS vulnerability on External links of the Project pages, by rendering the External links in Markdown (which auto-sanitises the URLs)

Fixes #5138

Staging branch: https://fix-xss.pfe-preview.zooniverse.org

screen shot 2018-12-11 at 16 59 48
_Screenshot shows an example project with an External Link that has JavaScript code as its URL. The <Markdown> element prevents the code from rendering properly._

screen shot 2018-12-11 at 17 01 39
_The external links on the example project, without the new fixes in this PR._

Minor side effects of this PR:
- Invalid URLs are no longer rendered as hyperlinks - it looks terrible when this happens, but it's safer than the current behaviour.
- Small style changes - note that the icons are no longer clickable.

### Primary Testing

Baseline comparison (without fixes in branch):
- go to https://master.pfe-preview.zooniverse.org/projects/darkeshard/transformers
- Find the external link on the Project Home page, titled "DECEPTICON HACK", which has the "URL": `javascript:alert('mwa ha ha')`
- Click on the link. The JavaScript `alert` should trigger, which indicates an XSS vulnerability. ☠️ (Users should not be able to run their own custom JavaScript code on their Project pages. Or anywhere, really.)

Fix confirmation:
- go to https://fix-xss.pfe-preview.zooniverse.org/projects/darkeshard/transformers
- Find the external link on the Project Home page, titled "DECEPTICON HACK". _It should not render as a hyperlink that can be clicked._
- The "link" should be unclickable, and the JavaScript `alert` should not be able to trigger. 👍  

### Basic Testing

- [x] Does the non-logged in home page render correctly?
- [x] Does the logged in home page render correctly?
- [x] Does the projects page render correctly?
- [x] Can you load project home pages?
- [x] Can you load the classification page?
- [x] Can you submit a classification?
- [ ] Does talk load correctly?
- [x] Can you post a talk comment?

### Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [ ] ~~If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?~~
- [ ] Are the tests passing locally and on Travis?

### Status
Priority patch. Ready for review. Tagging @eatyourgreens and @camallen

### CODE REVIEW

**Comment by eatyourgreens** [comment]
This can't be merged while the tests are failing.

**Comment by shaunanoordin** [comment]
Noted, thanks @eatyourgreens - I'm updating `ExternalLink.spec.jsx` now.

File: app/components/ExternalLinksBlock/components/ExternalLink/ExternalLink.jsx [hunk 1]
```diff
@@ -18,17 +17,18 @@ export default function ExternalLink({ className, isExternalLink, isSocialLink,
 
   if (isSocialLink) {
     const icon = socialIcons[site].icon;
-    linkProps['aria-label'] = socialIcons[site].ariaLabel;
```
[the remark this answers is not in the document]
Comment (amyrebecca) [reply app/components/ExternalLinksBlock/components/ExternalLink/ExternalLink.jsx]: i've made a note for this about follow-up

**Comment by eatyourgreens** [comment]
`expect(wrapper.find(Markdown).text()).to.equal(someMarkdownText)` should work with our current setup. We should come back to those tests. 

And the social links do need acccessible labels for the icons.

File: app/components/ExternalLinksBlock/components/ExternalLink/ExternalLink.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (eatyourgreens) [reply app/components/ExternalLinksBlock/components/ExternalLink/ExternalLink.jsx]: I think the original code was fine for social links but dangerous for external links. I suggest restoring the old code for social media icons but using the new code for external URLs.

**Pull request merged**